src/Cache/Engine/RedisEngine.php

```diff
@@ -105,6 +120,74 @@ public function init(array $config = []): bool
      * @return bool True if Redis server was connected
      */
     protected function _connect(): bool
+    {
+        if (!empty($this->_config['nodes'])) {
+            return $this->connectRedisCluster();
+        }
+
+        return $this->connectRedis();
+    }
+
+    /**
+     * Connects to a Redis cluster server
+     *
+     * @return bool True if Redis server was connected
+     */
+    protected function connectRedisCluster(): bool
+    {
+        $connected = false;
+
+        // @codeCoverageIgnoreStart
+        $ssl = [];
+        if ($this->_config['tls']) {
+            $map = [
+                'ssl_ca' => 'cafile',
+                'ssl_key' => 'local_pk',
+                'ssl_cert' => 'local_cert',
+                'verify_peer' => 'verify_peer',
+                'verify_peer_name' => 'verify_peer_name',
+                'allow_self_signed' => 'allow_self_signed',
+            ];
+
+            foreach ($map as $configKey => $sslOption) {
+                if (array_key_exists($configKey, $this->_config)) {
```

Comment:
not a big fan of slower array_key_exists() when the value shouldnt be null anyway.
Couldnt we just use isset() here?

Reply:
for TLS configuration, it's safest to respect the user's config even if it's empty or null